src/runtime/virtcontainers/sandbox.go

```diff
@@ -689,6 +689,12 @@ func (s *Sandbox) createResourceController() error {
 		//TODO: in Docker or Podman use case, it is reasonable to set a constraint. Need to add a flag
 		// to allow users to configure Kata to constrain CPUs and Memory in this alternative
 		// scenario. See https://github.com/kata-containers/runtime/issues/2811
+	} else {
```

Comment:
We've handled the `SingleContainer` case. Can we just fail here if spec is not found? Otherwise, I wonder how we can handle other spec-related operations.

Reply:
In fact, there are places we assume `GetPatchedOCISpec` can always find the spec. Can you please change `GetPatchedOCISpec` to default to the first container in the sandbox if no proper annotations are found, and change the callers accordingly? Let's make sure we have such a dependency applied.

Reply:
@bergwolf Sometimes it is not possible to use the first container in the sandbox as the OCI Spec, because some UTs use empty ContainerConfig to create sandboxes, and if they are modified, the default values ​​in most UTs will need to be changed. Here I am modifying to print debug information in the log instead of returning err. Use the sandbox id as the default cgroup path when the cgroupPath cannot be obtained from the OCI Spec.

Reply:
@yaoyinnan Let's not add meaningless code just for UTs. If UTs need to be fixed, let's fix them.